lib/auto-languageclient.ts

```diff
@@ -295,6 +295,25 @@ export default class AutoLanguageClient {
     return configuration
   }
 
+  /**
+   * (Optional) Determines the `languageId` string used for `textDocument/didOpen` notification. If {null} or
+   * {undefined} is returned, it will fall back to the {_editor}'s grammar name.
+   *
+   * You can override this like this:
+   *
+   *     class MyLanguageClient extends AutoLanguageClient {
+   *       getLanguageIdFromEditor(editor: TextEditor) {
+   *         if (editor.getGrammar().scopeName === "source.myLanguage") {
+   *           return "myCustumLanguageId"
+   *         }
+   *       }
+   *     }
+   *
+   * @param _editor A {TextEditor} which is opened.
+   * @returns A {string} of `languageId` used for `textDocument/didOpen` notification.
+   */
+  public getLanguageIdFromEditor(_editor: TextEditor): string | void | null {}
```

Comment:
I would prefer to have a working function and use it as it is. The client can override a custom behaviour if they want to.

```suggestion
  protected getLanguageIdFromEditor(_editor: TextEditor): string | void | null {
    return this._editor.getGrammar().name
  }
```